### PR TITLE
Updates the view for editing applications

### LINKED ISSUE #295: Edit application uses input field instead of selector on room

### PR DESCRIPTION
**Pull request by Paalar**
- Have updated applicationForm's -Preferences.tsx, -Personal.tsx, -Comments.tsx to allow for initializing values (text, checked, selected).
- Updated EditForm to use the previously mentioned component to update an application
- The states of EditForm ApplicationForm, as well as the previously mentioned components's keys and names now match the backend's model values for less confusion
- Updated outdated interface
- Resolves #295

### CODE REVIEW

File: frontend/src/components/ApplicationForm/ApplicationFormComments.tsx [hunk 1]
```diff
@@ -1,46 +1,65 @@
...
+  comments?: string;
+  isAdmin: boolean;
 }
 
+const title = <Undertittel>{_NEEDS_SUB_TITLE}</Undertittel>;
+
 export const ApplicationFormComments: React.FunctionComponent<IProps> = (props) => {
-  const { updateApplicationFormData } = props;
+  const { updateApplicationFormData, hasNeeds, isAdmin } = props;
+  const comments = props.comments ? props.comments : '';
```
Comment (aslakhol) [inline frontend/src/components/ApplicationForm/ApplicationFormComments.tsx]: Here you could use the more terse and readable: 
const comments = props.comments || '';

File: frontend/src/components/ApplicationForm/ApplicationFormComments.tsx [hunk 2]
```diff
@@ -1,46 +1,65 @@
...
+  const needsText = props.needs ? props.needs : '';
```
Comment (aslakhol) [inline frontend/src/components/ApplicationForm/ApplicationFormComments.tsx]: see above.

File: frontend/src/components/EditApplication/EditForm.tsx [hunk 3]
```diff
@@ -1,102 +1,102 @@
...
+  constructor(props: IProps) {
+    super(props);
+    const { application } = props;
+    const { user, partnerApplication, seatRollover, needs, comments, preferredRoom } = application;
+    let partner = '';
+    if (partnerApplication && partnerApplication.user) {
+      partner = partnerApplication.user.username;
+    }
+    this.state = {
+      comments: comments !== undefined ? comments : '',
```
Comment (aslakhol) [inline frontend/src/components/EditApplication/EditForm.tsx]: comments: comments || '',

File: frontend/src/components/EditApplication/EditForm.tsx [hunk 4]
```diff
@@ -1,102 +1,102 @@
...
+      email: user.email,
+      hasNeeds: needs !== undefined && needs.length > 0,
```
Comment (aslakhol) [inline frontend/src/components/EditApplication/EditForm.tsx]: see above.

File: frontend/src/components/EditApplication/EditForm.tsx [hunk 5]
```diff
@@ -1,102 +1,102 @@
...
+      hasPartner: partner.length > 0,
+      name: user.fullname,
+      needs: needs !== undefined ? needs : '',
```
Comment (aslakhol) [inline frontend/src/components/EditApplication/EditForm.tsx]: see above.

File: frontend/src/components/EditApplication/EditForm.tsx [hunk 6]
```diff
@@ -1,102 +1,102 @@
...
+      partnerUsername: partner,
+      preferredRoom: preferredRoom ? preferredRoom : '',
+      rank: application.rank,
+      seatRollover: seatRollover !== undefined ? seatRollover : false,
```
Comment (aslakhol) [inline frontend/src/components/EditApplication/EditForm.tsx]: see above.

File: frontend/src/components/EditApplication/EditForm.tsx (hunk 4 above)
Comment (Paalar) [reply frontend/src/components/EditApplication/EditForm.tsx]: Not possible it seems

File: frontend/src/components/ApplicationForm/ApplicationFormComments.tsx (hunk 1 above)
Comment (Paalar) [reply frontend/src/components/ApplicationForm/ApplicationFormComments.tsx]: fix'd

File: frontend/src/components/ApplicationForm/ApplicationFormComments.tsx (hunk 2 above)
Comment (Paalar) [reply frontend/src/components/ApplicationForm/ApplicationFormComments.tsx]: fix'd

**Comment by aslakhol** [comment]
css is still wack 
image

**Pull request merged**